public/posts/v1.7.md

```diff
@@ -0,0 +1,430 @@
+https://web.dev/fetch-upload-streaming/.
+
+## TLS session cache
+
+Some servers require that users reuse existing TLS sessions (for example FTP).
+Until now Deno has not had the ability to reuse TLS sessions, and instead would
+re-establish a new TLS session for each connection. In this release we have
+added a process global TLS session cache that will allows to reuse existing TLS
+sessions between the connections.
+
+The session cache is an in memory cache with a size of 1024 sessions. Overflow
+sessions will cause the oldest session to be evicted. The TLS session cache is
+used on a best effort basis.
+
+## Improvements to `deno lsp`
+
+Luca
+
+## Add support for utf-16 and big5 encoding
+
+Bartek
+
+## Changes to Deno APIs
+
+We are happy to stabilize `Deno.shutdown` in this release. This function lets
+you shut down a socket cleanly (so flushing all data before closing). This can
+be called instead of `Deno.close` for supported sockets. Unlike the unstable
+version of this API, no socket mode has to be specified when shutting down, as
+you only can only really shut down the write end of a socket.
+
+In this release the signature of the option bag for the unstable
+`Deno.createHttpClient` API has changed. This API can be used to customize how a
+fetch is executed. Instead of being able to specify `caFile` (the path to a file
+containing a custom CA certificate), you now specify `caData`. This means you
+can now use in memory certificates for `Deno.createHttpClient`.
+
+```diff
+- const client = Deno.createHttpClient({ caFile: "./my_ca.pem" });
+- const res = await fetch("https://my.kubernetes:4443", { client })
++ const client = Deno.createHttpClient({ caData: Deno.readFileSync("./my_ca.pem") });
++ const res = await fetch("https://my.kubernetes:4443", { client })
+```
+
+The unstable `Deno.permission` APIs for the `net` permission have also changed
+slightly. Instead of taking / returning a `url` parameter, we now use `host`, to
+match what the `--allow-net` flag supports.
+
+```diff
+- await Deno.permissions.query({ name: "net", url: "localhost:4000" });
++ await Deno.permissions.query({ name: "net", host: "localhost:4000" });
+```
+
+## Tokio 1.0
+
+Deno 1.7 marks the end of long migration from Tokio 0.2 to Tokio 1.0.
+
+We had a lot of issue reported by users about random hangs occuring in numerous
+`Deno` APIs. After thorough investigation it was determined that all of them
+were caused by interaction with the `tokio` runtime. Due to API changes in
+`tokio` 1.0 we had to rearchitecture significant parts of `deno_core` to cater
+for those changes. In effect the `ResourceTable`, a structure that holds Rust
+allocated objects (eg. a file handle, TCP connection) was rewritten from
+scratch. We had added the ability to queue different "ops" acting on the
+resources, as well as graceful closing of those resources. That means that any
+pending reads would be canceled when closing a resource, while pending writes
+run until completion.
```

Comment:
Additionally - assuming https://github.com/denoland/deno/pull/9118 gets landed before the release:

* TLS streams can now send and receive concurrently.
* TLS streams can now be closed gracefully.
* UDP and UnixDatagram sockets can now send and receive packets concurrently.
* TCP, TLS, UnixStream `write` operations are serialized w.r.t other `write` operations. Similarly, `read` operations are serialized w.r.t. other `read` ops.
*  All I/O performed on a `File` resource is serialized.

Reply:
Didn't make the cut, unfortunately.